File: jumpingOnClouds_hacker_rank.rb
def jumpingOnClouds(c)
    count = 0
    pos = 0
    while pos < c.length 
        if pos == c.length-1
            break
        end
        if(c[pos + 2] == 1)
            pos += 1
            count += 1
        else
            if pos + 1 == c.length
                count += 1
                pos += 1
                break
            else
                count += 1
                pos += 2
            end
        end
    end
    count
end

File: jumpingOnClouds_hacker_rank.js
function jumpingOnClouds(c) {
    let count = 0;
    let i = 0;
    while (i < c.length) {
        if (i == c.length - 2) {
            count += 1;
            break
        }
        if (c[i + 2] == 0) {
            i += 2;
            count += 1
        } else {
            i += 1;
            count += 1;
        }
    }
    return count;
}